src/parsemultipart.jl

```diff
@@ -0,0 +1,269 @@
+
+            if pair
+                i += 3
+            else
+                i += 1
+            end
+        else
+            i += 1
+        end
+    end
+    return retval
+end
+
+"""
+    parse_multipart_chunk(chunk)
+
+Parse a single multi-part chunk into a Multipart object.  This will decode
+the header and extract the contents from the byte array.
+"""
+function parse_multipart_chunk(chunk)
+    (startIndex, end_index) = find_header_boundary(chunk)
+
+    headers = String(view(chunk, startIndex:end_index))
+    content = view(chunk, end_index+1:lastindex(chunk))
+
+    disposition = match(r"(?i)Content-Disposition: form-data(.*)\r\n", headers)
+
+    if disposition === nothing
+        @warn "Content disposition is not specified dropping the chunk."
+        return # Specifying content disposition is mandatory
+    end
+
+    name = nothing
+    filename = nothing
+
+    for (pair, key, value) in content_disposition_extract(disposition[1])
+        if pair && key == "name"
+            name = value
+        elseif pair && key == "filename"
+            filename = value
+        end
+    end
+
+    name === nothing && return
+
+    match_contenttype = match(r"(?i)Content-Type: (\S*[^;\s])", headers)
+    contenttype = match_contenttype !== nothing ? match_contenttype[1] : "text/plain" # if content_type is not specified, the default text/plain is assumed
+
+    return Multipart(filename, IOBuffer(content), contenttype, "", name)
+end
+
+"""
+    parse_multipart_body(body, boundary)::Vector{Multipart}
+
+Parse the multipart body received from the client breaking it into the various
+chunks which are returned as an array of Multipart objects.
+"""
+function parse_multipart_body(body::AbstractVector{UInt8}, boundary::AbstractString)::Vector{Multipart}
+    multiparts = Multipart[]
+    idxs = find_multipart_boundaries(body, IOBuffer("--$(boundary)").data)
+    length(idxs) > 1 || (return multiparts)
+
+    for i in 1:length(idxs)-1
+        chunk = view(body, idxs[i][2]+1:idxs[i+1][1]-1)
+        push!(multiparts, parse_multipart_chunk(chunk))
+    end
+    return multiparts
+end
+
+
+"""
+    parse_multipart_form(req::Request)::Vector{Multipart}
+
+Parse the full mutipart form submission from the client returning and
+array of Multipart objects containing all the data.
+
+The order of the multipart form data in the request should be preserved.
+[RFC7578 5.2](https://tools.ietf.org/html/rfc7578#section-5.2).
+
+The boundary delimiter MUST NOT appear inside any of the encapsulated parts. Note
+that the boundary delimiter does not need to have '-' characters, but a line using
+the boundary delimiter will start with '--' and end in \r\n.
+[RFC2046 5.1](https://tools.ietf.org/html/rfc2046#section-5.1.1)
+"""
+function parse_multipart_form(req::Request)::Vector{Multipart}
+    # parse boundary from Content-Type
+    m = match(r"multipart/form-data; boundary=(.*)$", req["Content-Type"])
+    m === nothing && return nothing
```

Comment:
It seems like this will cause problems because you've set the function return type as `Vector{Multipart}`; you'll either need to change the return type to `::Union{Nothing, Vector{Multipart}}` or return `Multipart[]`.

Reply:
Also would probably be good to add a test-case for this specifically

Reply:
Updated the return type and added a test in 24edb33d.